src/Compilers/Extension/CompilerPackage.cs

```diff
@@ -0,0 +1,127 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.Shell;
+using System.Reflection;
+using System.IO;
+
+namespace Roslyn.Compilers.Extension
+{
+    [ProvideAutoLoad(UIContextGuids.SolutionExists)]
+    public sealed class CompilerPackage : Package
+    {
+        private const string WriteFileExceptionMessage =
+@"{1}
+
+To reload the Roslyn compiler package, close Visual Studio and any MSBuild processes, then restart Visual Studio.";
+
+        protected override void Initialize()
+        {
+            base.Initialize();
+
+            var packagePath = Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location);
+
+            string localRegistryRoot;
+            var reg = (ILocalRegistry2)this.GetService(typeof(SLocalRegistry));
+            reg.GetLocalRegistryRoot(out localRegistryRoot);
+            var registryParts = localRegistryRoot.Split('\\');
+
+            // Is it a valid Hive looks similar to:  
+            //  'Software\Microsoft\VisualStudio\14.0'  'Software\Microsoft\VisualStudio\14.0Roslyn'  'Software\Microsoft\VSWinExpress\14.0'
+            if (registryParts.Length >= 4)
+            {
+                var skuName = registryParts[2];
+                var hiveName = registryParts[3];
+                var roslynHive = string.Format(@"{0}.{1}", registryParts[2], registryParts[3]);
```

Comment:
I think there's better ways to fetch all of these values, so I left the code as is...but yeah, yeah...